app/controllers/queue_items_controller.rb

```diff
@@ -11,9 +11,20 @@ def create
     redirect_to my_queue_path
   end
 
+  def update_queue
+    begin
+      update_queue_items
+      current_user.normalize_queue_item_positions
+    rescue ActiveRecord::RecordInvalid
+      flash[:danger] = "Invalid position number."
+    end
+    redirect_to my_queue_path
+  end
+
   def destroy
     queue_item = QueueItem.find(params[:id])
     queue_item.destroy if current_user.queue_items.include?(queue_item)
```

Comment:
This part should be refactored to model - `current_user.queue_items.include?(queue_item)`